### PR TITLE
Fixed issue #72

### PR DESCRIPTION
**Pull request by danielhaviv**

### CODE REVIEW

File: python/deltalake/__init__.py
Comment (houqp) [inline python/deltalake/__init__.py:34]: for local files, self._table.file_paths() is invoked twice, could you save it as a local var at the beginning of the function and reuse that local var here?

File: python/deltalake/__init__.py
Comment (houqp) [inline python/deltalake/__init__.py:32]: Good practice to avoid string concatenation using `+`

```suggestion
            return dataset(keys, filesystem=f"{paths[0].scheme}://{paths[0].netloc}")
```

**Review by houqp — APPROVED** [review]
Good catch :+1:

File: python/deltalake/__init__.py
Comment (danielhaviv) [reply python/deltalake/__init__.py:34]: Done

**Comment by rtyler** [comment]
The build error looks to be unrelated, and might be a similar problem to that which is described in #69 but manifesting through our integration tests.

```
running 2 tests
test s3::test_s3_simple_golden ... ok
test s3::test_s3_simple ... FAILED

failures:

---- s3::test_s3_simple stdout ----
thread 's3::test_s3_simple' panicked at 'dispatch dropped without returning error', /home/tyler/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.13.9/src/client/conn.rs:234:35
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```


I'll let @houqp merge this if he's satisfied with the changes.

**Comment by houqp** [comment]
#69 is a bug in python binding, while this test is rust only, so it's something else. I will merge this PR for now. Integration test should be fixed in a separate PR.

**Comment by houqp** [comment]
Thanks @danielhaviv for the contribution!

**Comment by danielhaviv** [comment]
What are the release cycles (or when should I expect to see the fix in PyPi)?

**Comment by houqp** [comment]
we don't have a set release cadence, I was hoping to cut a new release after #75 gets merged.

**Pull request merged**
